src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,143 @@
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.StatorCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.motorcontrol.can.TalonSRXConfiguration;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.UnitModel;
+
+/**
+ * Add angle motor, drive motor, unitModelVelocity and unitModel.
+ */
+public class SwerveModule extends SubsystemBase {
+    public WPI_TalonSRX angleMotor;
+    public WPI_TalonFX driveMotor;
+    private final UnitModel unitModelVelocity = new UnitModel(Constants.Swerve.VELOCITY_TICKS_PER_UNIT);
+    private final UnitModel unitModelDegree = new UnitModel(Constants.Swerve.DEGREE_TICKS_PER_UNIT);
+    private int i;
+
+    public SwerveModule(int i) {
+        this.i = i;
+
+        /**
+         * Add PID, encoder, Voltage Compensation, Current Limit.
+         */
+        this.angleMotor = new WPI_TalonSRX(Constants.Swerve.ANGLE_MOTOR[i]);
+        angleMotor.config_kP(0, Constants.Swerve.ANGLE_MOTOR_PID_P[i]);
+        angleMotor.config_kI(0, Constants.Swerve.ANGLE_MOTOR_PID_I[i]);
+        angleMotor.config_kD(0, Constants.Swerve.ANGLE_MOTOR_PID_D[i]);
+        this.angleMotor.configSelectedFeedbackSensor(FeedbackDevice.Analog, 0, 10);
+        angleMotor.enableVoltageCompensation(true);
+        angleMotor.configPeakCurrentLimit(Constants.Swerve.CURRENT_LIMIT, 10);
+
+        /**
+         * Add PID, encoder, Closed loop Ramp, Voltage Compensation, CurrentLimit.
+         */
+        this.driveMotor = new WPI_TalonFX(Constants.Swerve.DRIVE_MOTOR[i]);
+        driveMotor.config_kP(0, Constants.Swerve.ANGLE_MOTOR_PID_P[i]);
+        driveMotor.config_kI(0, Constants.Swerve.ANGLE_MOTOR_PID_I[i]);
+        driveMotor.config_kD(0, Constants.Swerve.ANGLE_MOTOR_PID_D[i]);
+        this.driveMotor.configSelectedFeedbackSensor(FeedbackDevice.IntegratedSensor, 0, 10);
+        driveMotor.configClosedloopRamp(Constants.Swerve.RAMP_RATE);
+        driveMotor.enableVoltageCompensation(true);
+        driveMotor.configStatorCurrentLimit(new StatorCurrentLimitConfiguration(true, Constants.Swerve.CURRENT_LIMIT, Constants.Swerve.TRIGGER_CURRENT, Constants.Swerve.TRIGGER_TIME), 10);
+    }
+
+    /**
+     * Find the smallest error.
+     * @param angle
+     * @param currentAngle
+     * @return
+     */
+    public static double getTargetError(double angle, double currentAngle) {

```

Comment:
I'm not sure about that, but my intuition tells me that you can combine all cases into one calculation. Try to think about that 🙂.